common/json.hpp

```diff
@@ -18,23 +18,85 @@
 #include <common/error.hpp>
 #include <common/expected.hpp>
 
-namespace json {
+#include "config.h"
+
+#ifdef MENDER_COMMON_USE_NLOHMANN_JSON
+#include <nlohmann/json.hpp>
+#endif
+
+using namespace std;
+
+namespace mender::common::json {
 
 enum JsonErrorCode {
 	NoError = 0,
-	KeyNoExist,
+	ParseError,
+	KeyError,
+	IndexError,
 	TypeError,
 };
 using JsonError = mender::common::error::Error<JsonErrorCode>;
 
 class Json {
 public:
 	using ExpectedJson = mender::common::expected::Expected<Json, JsonError>;
```

Comment:
According to the google style guide, the `using ` namespace includes are no go:
https://google.github.io/styleguide/cppguide.html#Namespaces

Reply:
I *think* they mean only at a global level though. They don't mention it explicitly, but they use `using` in their own example, inside a namespace.